Doppler.Sap.Job/DopplerCurrencyService/DopplerCurrencyService.cs

```diff
@@ -1,6 +1,82 @@
-﻿namespace Doppler.Sap.Job.Service.DopplerCurrencyService
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Doppler.Sap.Job.Service.Dtos;
+using Doppler.Sap.Job.Service.Logger;
+using Doppler.Sap.Job.Service.Settings;
+using Newtonsoft.Json;
+
+namespace Doppler.Sap.Job.Service.DopplerCurrencyService
 {
-    public class DopplerCurrencyService
+    public class DopplerCurrencyService : IDopplerCurrencyService
     {
+        private readonly HttpClient _httpClient;
+        private readonly DopplerCurrencySettings _dopplerCurrencySettings;
+        private readonly ILoggerAdapter<DopplerCurrencyService> _logger;
+
+        public DopplerCurrencyService(
+            IHttpClientFactory httpClientFactory,
+            HttpClientPoliciesSettings httpClientPoliciesSettings,
+            DopplerCurrencySettings dopplerCurrencySettings,
+            ILoggerAdapter<DopplerCurrencyService> logger)
+        {
+            _httpClient = httpClientFactory.CreateClient(httpClientPoliciesSettings.ClientName);
+            _dopplerCurrencySettings = dopplerCurrencySettings;
+            _logger = logger;
+        }
+
+        public async Task<IList<CurrencyDto>> GetCurrencyByCode()
+        {
+            var dateTimeNow = DateTime.UtcNow;
+            var returnList = new List<CurrencyDto>();
+            
+            foreach (var currencyCode in _dopplerCurrencySettings.CurrencyCodeList)
+            {
+                var uri = new Uri(_dopplerCurrencySettings.Url + $"{currencyCode}/{dateTimeNow.Year}-{dateTimeNow.Month}-{dateTimeNow.Day}");
```

Comment:
Is it timezone safe? I mean you do not know the time zone of the server that runs it.

Reply:
You are right, we use the Timezone set in `appsettings.json` to get `DateTime.UtcNow`, the fix is implemented in this commit https://github.com/MakingSense/doppler-jobs/pull/1/commits/3bf4b10bdadf4513809a8e4f691d94481449b369

Reply:
Mmmm, if I am not wrong, that setting is not related to this.

Reply:
Oh, I understand now, thanks!